### PR TITLE
[BTCV2-1770] CodePush Improvements

### PR DESCRIPTION
**Pull request by mlcbest**
## What does this PR do?
Make codepush work only in production mode. Separate Sentry source maps versions from OTA updates versions

#### *Any background context you want to provide?*
Only production version

#### *Any scout cleaning/refactoring?*
Refactored Appdelegate Objective-C code to work properly 

## Todos:

- [+ ] Checked on iOS
- [+ ] Checked on Android

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import * as Sentry from '@sentry/react-native';
-import React from 'react';
+import React, { Component } from 'react';
 import { I18nextProvider } from 'react-i18next';
-import { View, StyleSheet, LogBox } from 'react-native';
+import { View, StyleSheet, LogBox, Platform } from 'react-native';
 import codePush from 'react-native-code-push';
+import Config from 'react-native-config';
```
Comment (PanRadaCloudbest) [inline App.tsx:6]: hmm can we use global config  from folder config in src ?

File: App.tsx [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import * as Sentry from '@sentry/react-native';
-import React from 'react';
+import React, { Component } from 'react';
 import { I18nextProvider } from 'react-i18next';
-import { View, StyleSheet, LogBox } from 'react-native';
+import { View, StyleSheet, LogBox, Platform } from 'react-native';
```
Comment (PanRadaCloudbest) [inline App.tsx:4]: Platform we have in helpers

File: ios/GoldWallet.xcodeproj/xcshareddata/xcschemes/GoldWallet (Debug).xcscheme [hunk 3]
```diff
@@ -10,7 +10,7 @@
             ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
             <ActionContent
                title = "Run Script"
-               scriptText = "cp &quot;${PROJECT_DIR}/../.env.prod.mainnet&quot; &quot;${PROJECT_DIR}/../.env&quot;&#10;cp -r &quot;${PROJECT_DIR}/GoogleService-Info-prod.plist&quot; &quot;${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app/GoogleService-Info.plist&quot;&#10;">
+               scriptText = "cp &quot;${PROJECT_DIR}/../.env.dev.testnet&quot; &quot;${PROJECT_DIR}/../.env&quot;&#10;cp -r &quot;${PROJECT_DIR}/GoogleService-Info-prod.plist&quot; &quot;${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app/GoogleService-Info.plist&quot;&#10;">
```
Comment (PanRadaCloudbest) [inline ios/GoldWallet.xcodeproj/xcshareddata/xcschemes/GoldWallet (Debug).xcscheme:13]: ?? why it is production target and must prod.mainet

Comment (mlcbest) [reply ios/GoldWallet.xcodeproj/xcshareddata/xcschemes/GoldWallet (Debug).xcscheme:13]: ahh, ok, I thought it is debug === development :)

File: App.tsx (hunk 2 above)
Comment (mlcbest) [reply App.tsx:4]: fixed

File: App.tsx (hunk 1 above)
Comment (mlcbest) [reply App.tsx:6]: fixed

**Review by aleparty — APPROVED** [review]

**Review by PanRadaCloudbest — APPROVED** [review]

**Pull request merged**